orchagent/vnetorch.cpp

```diff
@@ -848,6 +848,162 @@ bool VNetRouteOrch::removeNextHopGroup(const string& vnet, const NextHopGroupKey
     return true;
 }
 
+bool VNetRouteOrch::createNextHopGroup(const string& vnet,
+                                       NextHopGroupKey& nexthops,
+                                       VNetVrfObject *vrf_obj,
+                                       const string& monitoring)
+{
+
+    if (nexthops.getSize() == 0)
+    {
+        return true;
+    }
+    else if (nexthops.getSize() == 1)
+    {
+        NextHopKey nexthop(nexthops.to_string(), true);
+        NextHopGroupInfo next_hop_group_entry;
+        next_hop_group_entry.next_hop_group_id = vrf_obj->getTunnelNextHop(nexthop);
+        next_hop_group_entry.ref_count = 0;
+        if (monitoring == "custom" || nexthop_info_[vnet].find(nexthop.ip_address) == nexthop_info_[vnet].end() || nexthop_info_[vnet][nexthop.ip_address].bfd_state == SAI_BFD_SESSION_STATE_UP)
+        {
+            next_hop_group_entry.active_members[nexthop] = SAI_NULL_OBJECT_ID;
+        }
+        syncd_nexthop_groups_[vnet][nexthops] = next_hop_group_entry;
+    }
+    else
+    {
+        if (!addNextHopGroup(vnet, nexthops, vrf_obj, monitoring))
+        {
+            SWSS_LOG_ERROR("Failed to create next hop group %s", nexthops.to_string().c_str());
+            return false;
+        }
+    }
+    return true;
+}
+
+NextHopGroupKey VNetRouteOrch::createActiveNHSet(const string& vnet,
+                                       NextHopGroupKey& nexthops,
+                                       IpPrefix& ipPrefix)
+{
+    NextHopGroupKey nhg_custom("", true);
+    set<NextHopKey> next_hop_set = nexthops.getNextHops();
+    for (auto it : next_hop_set)
+    {
+        if(monitor_info_.find(vnet) != monitor_info_.end() &&
+            monitor_info_[vnet].find(ipPrefix) != monitor_info_[vnet].end())
+        {
+            for (auto monitor : monitor_info_[vnet][ipPrefix])
+            {
+                if (monitor.second.endpoint == it)
+                {
+                    if (monitor.second.state == MONITOR_SESSION_STATE_UP)
+                    {
+                        // monitor session exists and is up
+                        nhg_custom.add(it);
+
+                    }
+                    continue;
+                }
+            }
+        }
+    }
+    return nhg_custom;
+}
+
```

Comment:
Please add a high level comment on what this function does and what are the input/output parameters.

Reply:
done